### PR TITLE
feature/enable sending qml ui over network

### PR DESCRIPTION
**Pull request by AIIX**
## Description
Enables sending qml ui from remote mycroft-core instance over network 

## How to test
Add to mycroft.conf (any level):
{
    "remote": true,
    "remote-server": "https://instance-where-mycroft-core-is-hosted-.com"
}

Additional Configuration Server Side Running Apache:
`ln -s /opt /var/www/html/opt`

## Contributor license agreement signed?
CLA [Yes] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/skills/core.py [hunk 1]
```diff
@@ -234,6 +234,11 @@ def __init__(self, skill):
         self.page = None    # the active GUI page (e.g. QML template) to show
         self.skill = skill
         self.on_gui_changed_callback = None
+        self.config_core = Configuration.get()
```
Comment (forslund) [inline mycroft/skills/core.py]: let's just call this member `config`. I know it doesn't match the one in the MycroftSkill class but that will be changed from self.config_core to self.config as part of 19.08

File: mycroft/skills/core.py [hunk 2]
```diff
@@ -234,6 +234,11 @@ def __init__(self, skill):
...
+        
+    @property
+    def remote_url(self):
```
Comment (forslund) [inline mycroft/skills/core.py]: Add simple docstring here.

**Comment by AIIX** [comment]
Yes, the generated path would be: http://my-mycroft.com/opt/mycroft/skills/mycroft-weather.mycroftai/ui/weather.qml, the simplest way i did it was linking /opt/mycroft/skills to the default html folder, access level of directory could be changed to only be able to view the skill folder from apache but the path would remain the same as above, or with a bit of tweaking path could be different. 

Fixing rest of the issues.

**Comment by forslund** [comment]
Works nicely as far as I can tell. Merging!

In the future we might want to add a root_directory config as well so a server root can be in /opt/mycroft or something like that.

**Pull request merged**